test/cpp/tensorexpr/test_dynamic_shapes.cpp

```diff
@@ -462,5 +462,85 @@ TEST(DynamicShapes, GraphWithCatAndBroadcast) {
+          %input.4 : Long(SS(-2), SS(-6), requires_grad=0, device=cpu),
+          %4 : Float(SS(-7), requires_grad=0, device=cpu),
+          %5 : Float(SS(-7), requires_grad=0, device=cpu),
+          %SS_10 : int,
+          %SS_9 : int,
+          %SS_8 : int,
+          %SS_7 : int,
+          %SS_6 : int,
+          %SS_5 : int,
+          %SS_4 : int,
+          %SS_3 : int,
+          %SS_2 : int):
+      %15 : int = prim::Constant[value=1]()
+      %16 : bool = prim::Constant[value=0]()
+      %17 : int = prim::Constant[value=6]()
+      %18 : Float(SS(-2), SS(-6), strides=[139, 1], requires_grad=0, device=cpu) = aten::to(%input.4, %17, %16, %16)
+      %19 : Tensor[] = prim::ListConstruct(%0, %1, %18, %2)
+      %20 : Float(SS(-2), SS(-8), strides=[261, 1], requires_grad=0, device=cpu) = aten::cat(%19, %15)
+      %21 : Float(SS(-2), SS(-9), strides=[261, 1], requires_grad=0, device=cpu) = aten::add(%20, %5, %15)
+      %22 : Float(SS(-2), SS(-10), requires_grad=0, device=cpu) = aten::mul(%21, %4)
+      return (%22))IR";
+  parseIR(graph_string, &*graph);
+
+  std::vector<torch::jit::StrideInput> input_desc = {
+      torch::jit::StrideInput::TENSOR_CONT};
+  std::unordered_map<
+      const torch::jit::Value*,
+      std::vector<torch::jit::StrideInput>>
+      symbolic_strides;
+  symbolic_strides[graph->inputs().at(0)] = input_desc;
+  symbolic_strides[graph->inputs().at(1)] = input_desc;
+  symbolic_strides[graph->inputs().at(2)] = input_desc;
+  symbolic_strides[graph->inputs().at(3)] = input_desc;
+  symbolic_strides[graph->inputs().at(4)] = input_desc;
+  symbolic_strides[graph->inputs().at(5)] = input_desc;
+  symbolic_strides[graph->outputs().at(0)] = input_desc;
+  std::vector<int64_t> symbolic_shape_inputs = {
+      -10, -9, -8, -7, -6, -5, -4, -3, -2};
+  TensorExprKernel k(graph, {}, symbolic_shape_inputs, false, symbolic_strides);
+
+  auto stmt = k.getCodeGenStmt();
+
+  int64_t i2 = 10;
+  int64_t i3 = 32;
+  int64_t i4 = 19;
+  int64_t i5 = 71;
+  int64_t i6 = 139;
+  int64_t i7 = 261;
+  int64_t i8 = 261;
+  int64_t i9 = 261;
+  int64_t i10 = 261;
+  auto x0 = at::rand({10, 32}, at::TensorOptions(at::kCPU).dtype(at::kFloat));
+  auto x1 = at::rand({10, 19}, at::TensorOptions(at::kCPU).dtype(at::kFloat));
+  auto x2 = at::rand({10, 71}, at::TensorOptions(at::kCPU).dtype(at::kFloat));
+  auto x3 = at::ones({10, 139}, at::TensorOptions(at::kCPU).dtype(at::kLong));
+  auto x4 = at::rand({261}, at::TensorOptions(at::kCPU).dtype(at::kFloat));
+  auto x5 = at::rand({261}, at::TensorOptions(at::kCPU).dtype(at::kFloat));
+  std::vector<at::Tensor> inputs = {x0, x1, x2, x3, x4, x5};
+
+  std::vector<IValue> stack = at::fmap<at::IValue>(inputs);
+  stack.push_back(i10);
+  stack.push_back(i9);
+  stack.push_back(i8);
+  stack.push_back(i7);
+  stack.push_back(i6);
+  stack.push_back(i5);
+  stack.push_back(i4);
+  stack.push_back(i3);
+  stack.push_back(i2);
+  k.run(stack);
```

Comment:
did we want to use the  other path  as well ? pre-allocated outputs?

Reply:
I didn't add it earlier since I had that PR in a different stack. Now that has landed. So, added a check using that as well.